plugins/out_s3/s3.c

```diff
@@ -118,6 +118,56 @@ static char *mock_error_response(char *error_env_var)
     return NULL;
 }
 
+static void s3_retry_warn(struct flb_s3 *ctx, const char *tag,
+                          char *input_name, time_t create_time,
+                          int less_than_limit)
+{
+    struct tm now_time;
+    char create_time_str[20];
+    struct tm *tmp;
+
+    tmp = localtime_r(&create_time, &now_time);
+    strftime(create_time_str, 20, "%Y-%m-%d %H:%M:%S", tmp);
+    if (input_name == NULL || strlen(input_name) == 0) {
+        if (less_than_limit == FLB_TRUE) {
+            flb_plg_warn(ctx->ins,
+                        "failed to flush chunk tag=%s, create_time=%s, "
+                        "retry issued: (out_id=%d)",
+                        tag, create_time_str, ctx->ins->id);
+        }
+        else {
+            flb_plg_warn(ctx->ins,
+                        "chunk tag=%s, create_time=%s cannot be retried",
+                        tag, create_time_str);
+        }   
+    }
+    else {
+        if (less_than_limit == FLB_TRUE) {
+            flb_plg_warn(ctx->ins,
+                        "failed to flush chunk tag=%s, create_time=%s, "
+                        "retry issued: input=%s > output=%s (out_id=%d)",
+                        tag, create_time_str, input_name, ctx->ins->name, ctx->ins->id);
+        }
+        else {
+            flb_plg_warn(ctx->ins,
+                        "chunk tag=%s, create_time=%s cannot be retried: "
+                        "input=%s > output=%s",
+                        tag, create_time_str, input_name, ctx->ins->name);
+        }
+    }
+}
+
+static void s3_decrement_index(struct flb_s3 *ctx)
+{
+    int ret;
+    ctx->seq_index--;
+
+    ret = write_seq_index(ctx->seq_index_file, ctx->seq_index);
+    if (ret < 0) {
+        flb_plg_error(ctx->ins, "Failed to decrement $INDEX for s3 key after request error");
```

Comment:
I realized we should be more clear in this message. We did successfully decrement it, we didn't save it to the store_dir.

So something like:

"Failed to save decremented $INDEX for s3 key to store_dir after request error"